promote-release-to-box.py

```diff
@@ -0,0 +1,290 @@
+        --artifactory-url https://repo.orl.eng.hitachivantara.com/artifactory \
+        --password --user buildguy
+
+    2. jf rt search --server-id orl-artifactory \
+        --props "build.name=pdi-xxx-9.5.1.0;build.number=86" "*-9.5.1.0-86.zip" \
+        > builds.json
+
+    Alternative way: Use ArtifactoryBuildManager https://github.com/devopshq/artifactory#builds
+    
+    ^^^^^^^^^^^^^
+    from artifactory import ArtifactoryBuildManager
+
+    arti_build = ArtifactoryBuildManager(
+        orl_url, auth=auth_orl)
+
+    # Get all builds,space turns into %
+    all_builds = arti_build.builds
+    print(all_builds)
+    ^^^^^^^^^^^^^
+
+    '''
+    
+
+    # Define the command and arguments
+    command = [
+        'jf', 'config', 'add', 'orl-artifactory',
+        '--interactive=false', '--enc-password=false', '--basic-auth-only',
+        '--artifactory-url', f'{rt_base_url}',
+        '--password', f'{rt_auth[0]}',
+        '--user', f'{rt_auth[1]}'
+    ]
+
+    # Execute the command
+    subprocess.run(command)
+    
+    # Define the command and arguments
+    command = ['jf', 'rt', 'search', '--server-id', 'orl-artifactory', '--props', 
+               f'build.name={build_name};build.number={build_number}',
+               f'*' ]
+               # f'{artifact_name}']
+    
+    output_file = 'artifacts.json'
+
+    # Execute the command and capture the output
+    result = subprocess.run(command, capture_output=True, text=True)
+
+    # Parse the command output as JSON
+    output_json = json.loads(result.stdout)
+
+    # Save the JSON object to a file
+    with open(output_file, 'w') as file:
+        json.dump(output_json, file, indent=4)
+    
+    return output_json, set([artifact['path'].split('/')[-1] for artifact in output_json])
+
+
+def download_artifacts_v3(artifacts_to_release, builds_output_json, auth=(None, None), rt_base_url = None):
+    release_artifact_downloaded = []
+    for build_artifact in tqdm(builds_output_json):
+        file_name = build_artifact['path']
+        if build_artifact['path'].split('/')[-1] in artifacts_to_release.keys():
+
+            logging.info(f"Downloading {file_name}")
+
+            path = build_artifact['path']
+            sha1 = build_artifact['sha1']
+            sha256 = build_artifact['sha256']
+            md5 = build_artifact['md5']
+
+            file_name = build_artifact['path'].split('/')[-1]
+            check_sum_file_name = path.split('/')[-1]+'.sum'
+
+            # download artifact
+            rt_path = ArtifactoryPath(
+                f"{rt_base_url}/"+path, auth=auth, auth_type=HTTPBasicAuth
+            )
+            
+            if not os.path.exists(file_name):
+                logging.info(f'Downloading {file_name} from {rt_path}.')
+                rt_path.writeto(out=file_name, progress_func=None)
+                logging.info(f'Download complete.')
+            
+            if not os.path.exists(check_sum_file_name):
```

Comment:
if the checksum file doesn't exist yet, then you create it.

Reply:
yes